ext/java/org/msgpack/jruby/Encoder.java

```diff
@@ -38,13 +38,16 @@
   private final Encoding utf8Encoding;
   private final boolean compatibilityMode;
   private final ExtensionRegistry registry;
+  private final Packer packer;
 
   public boolean hasSymbolExtType;
   private boolean hasBigintExtType;
+  private boolean streaming;
```

Comment:
It's not clear to me what "streaming" means for now. In the serialization/deserialization context, "streaming" means a streaming encoder/decoder like this:
```
decoder = Decoder.new
decoder.register_callback do |decoded_object|
  # ...
end
decoder << str1
decoder << str2
decoder << str3
```

Reply:
I'm not particularly happy with this name either. `streaming` is meant to convey that at this stage the size isn't know yet, the extension callback will "stream" some bytes, that we'll measure once it's done.

If you have idea of a better name, let me know i"ll happily rename it.

Reply:
How about `swapped` or `temp_buffer`?

Reply:
Maybe `recursive`?

On a similar note we should probably pass a keyword argument rather than rely on arity: https://github.com/msgpack/msgpack-ruby/pull/261#discussion_r824519919, so we should probably select a user facing name for this concept.

Reply:
A longer name sounds good to me if we'll use it for a keyword argument :P
`recursive_extension_handling` ... hmm, not beautiful.

Reply:
`recursive` would look like this:
```ruby
        factory.register_type(
          0x01,
          Point,
          packer: ->(point, packer) { packer.write(point.to_h) },
          unpacker: ->(unpacker) { Point.new(unpacker.read) },
          recursive: true,
        )
```

Since it's an argument to `register_type`, it's implied that it means `recursive_type`.

We could use `recursiveExtensionType` internally?

Reply:
Or just `recursiveExtension`?

Reply:
> Since it's an argument to `register_type`, it's implied that it means `recursive_type`.

👍 

> We could use `recursiveExtensionType` internally?

👍

Reply:
Ok, doing that then.